src/forder.c

```diff
@@ -1248,17 +1248,21 @@ SEXP isOrderedSubset(SEXP x, SEXP nrow)
 // specialized for use in [.data.table only
 // Ignores 0s but heeds NAs and any out-of-range (which result in NA)
```

Comment:
what was designed behaviour for NA based on that comment? it it will return FALSE for any NA, which solves the issue, and does not break any tests

Reply:
The behaviour intended was to heed NAs; meaning NAs anywhere should return FALSE.  It was getting it wrong then when there is one NA at the beginning.  Good spot and fix.